### PR TITLE
Bug fix for gmin and blank strings

### LINKED ISSUE #4848: Bug in gmin and blank strings
min of character with blank strings can give wrong results:
```R
> library(data.table)
> d = data.table(id=c(1,1), a=c("","a"))
> d[, min(a), id]
      id     V1
   <num> <char>
1:     1      a
> data.table:::gmin(c("a", ""))
[1] ""
> data.table:::gmin(c("", "a"))
[1] "a"
```
Expected behavior:
```R
> d[, base::min(a), id]
      id     V1
   <num> <char>
1:     1
> min(c("a", ""))
[1] ""
> min(c("", "a"))
[1] ""
```
A possible fix is in #4847 .

<details>
  <summary>Output of sessionInfo()</summary>
R version 3.6.3 (2020-02-29)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 20.04.1 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.9.0
LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.9.0

locale:
 [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8
 [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8
 [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C
[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base

other attached packages:
[1] data.table_1.13.5

loaded via a namespace (and not attached):
[1] compiler_3.6.3
</details>

### PR DESCRIPTION
**Pull request by khotilov**
Closes #4848 

Before the fix:
```R
> library(data.table)
> d = data.table(id=c(1,1), a=c("","a"))
> d[, min(a), id]
      id     V1
   <num> <char>
1:     1      a
> data.table:::gmin(c("a", ""))
[1] ""
> data.table:::gmin(c("", "a"))
[1] "a"
```
After the fix:
```R
> d = data.table(id=c(1,1), a=c("","a"))
> d[, min(a), id]
      id     V1
   <num> <char>
1:     1
> data.table:::gmin(c("a", ""))
[1] ""
> data.table:::gmin(c("", "a"))
[1] ""
```

### CODE REVIEW

**Comment by MichaelChirico** [comment]
thanks! could you please file an Issue to go with this?

please include in the issue the base behavior for this case as well

**Comment by khotilov** [comment]
@MichaelChirico here's an issue #4848

File: src/gsumm.c [hunk 1]
```diff
@@ -726,22 +726,26 @@ SEXP gmin(SEXP x, SEXP narm)
     break;
   case STRSXP:
     ans = PROTECT(allocVector(STRSXP, ngrp)); protecti++;
+    for (i=0; i<ngrp; i++) SET_STRING_ELT(ans, i, NA_STRING);
     if (!LOGICAL(narm)[0]) {
-      for (i=0; i<ngrp; i++) SET_STRING_ELT(ans, i, R_BlankString);
+      Rboolean *upd = calloc(ngrp, sizeof(Rboolean));
+      if (!upd) error(_("Unable to allocate %d * %d bytes for the update mask in gmin na.rm=FALSE"), ngrp, sizeof(Rboolean));
```
Comment (MichaelChirico) [inline src/gsumm.c:732]: is upd for update?

File: src/gsumm.c [hunk 2]
```diff
@@ -726,22 +726,26 @@ SEXP gmin(SEXP x, SEXP narm)
     break;
   case STRSXP:
     ans = PROTECT(allocVector(STRSXP, ngrp)); protecti++;
+    for (i=0; i<ngrp; i++) SET_STRING_ELT(ans, i, NA_STRING);
     if (!LOGICAL(narm)[0]) {
-      for (i=0; i<ngrp; i++) SET_STRING_ELT(ans, i, R_BlankString);
+      Rboolean *upd = calloc(ngrp, sizeof(Rboolean));
```
Comment (MichaelChirico) [inline src/gsumm.c:731]: this should be PROTECT(allocateVector(LGLSXP, ngrp)) IINM. then UNPROTECT(1) instead of free below

Comment (jangorecki) [reply src/gsumm.c:731]: Rboolean is not LGLSXP so not need protect, it is just int of 0,1,INT_MIN. I prefer to use bool instead.

File: src/gsumm.c (hunk 1 above)
Comment (khotilov) [reply src/gsumm.c]: Yep, changed to `updated`.

File: src/gsumm.c (hunk 2 above)
Comment (khotilov) [reply src/gsumm.c]: @jangorecki  I've just changed it to bool. The INT_MIN is not happening here.

**Comment by mattdowle** [comment]
Thanks @khotilov. I've invited you to be project member so that, among other things, you can create branches directly in the main project in future. The invite is a button that appears in your GitHub profile or projects page which you need to click to accept. Also added you to DESCRIPTION in this PR. Thanks again, and welcome!

**Pull request merged**